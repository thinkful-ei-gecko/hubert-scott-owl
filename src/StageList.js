import React from 'react';
import Stage from './Stage';

export default function StageList(props) {
    let stageUsers = props.stage.filter(users => {users.onStage})
    stageUsers = stageUsers.map(users => <Stage key={users.id} name={users.name} avatar={users.avatar}/>)

    return (
        <div className="stage" key={props.id}>
            {stageUsers}
        </div>
    )
}
