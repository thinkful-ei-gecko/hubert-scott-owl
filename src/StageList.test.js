import React from 'react';
import ReactDOM from 'react-dom';
import StageList from './StageList';
import renderer from 'react-test-renderer';
import { exportAllDeclaration } from '@babel/types';

describe('<StageList />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<StageList stage = {    [{
            id: 1,
            name: 'Koren Templeton',
            avatar:
                'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        },
        {
            id: 2,
            name: 'Caty Flucker',
            avatar:
                'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: false
        }
        ]} />, div);

        ReactDOM.unmountComponentAtNode(div);
    });
});

describe('<StageList />', () => {
    it('renders UI correctly', () => {
        const tree = renderer.create(<StageList stage = {    [{
            id: 1,
            name: 'Koren Templeton',
            avatar:
                'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        },
        {
            id: 2,
            name: 'Caty Flucker',
            avatar:
                'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: false
        }
    ]} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
