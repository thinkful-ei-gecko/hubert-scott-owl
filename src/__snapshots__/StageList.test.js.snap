// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageList /> renders UI correctly 1`] = `
<div
  className="stage"
/>
`;
